src/node_jsvmapi.cc

```diff
@@ -465,22 +465,58 @@ namespace v8impl {
 
 }  // end of namespace v8impl
 
-// We define a macro here because, with full error handling, we may want to
-// perform additional actions at the top of our public APIs, such as bail,
-// returning an error code at the very top of each public API in the case that
-// there's already an exception pending in v8impl::TryCatch::lastException
+   // We define a macro here because, with full error handling, we may want to
+   // perform additional actions at the top of our public APIs, such as bail,
+   // returning an error code at the very top of each public API in the case that
+   // there's already an exception pending in v8impl::TryCatch::lastException
 #define NAPI_PREAMBLE(e) \
```

Comment:
I like this macro. Also not sure about using an enum or bit flag for the pending exception, though. Should we always clear the error code here or clear it when we return napi_ok from the API (convert all returns from API to first set the error code before returning it) ? 